plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -388,11 +434,234 @@ public synchronized Optional<ConnectorOutputMetadata> finishCreateTable(
+        }).collect(toImmutableList()));
+    }
+
+    private synchronized void createViewWithPredicates(ConnectorSession session, SchemaTableName tableName, TableInfo info, Collection<ComputedStatistics> computedStatistics)
+    {
+        Map<String, Object> minimums = new LinkedHashMap<>();
+        Map<String, Long> distinctValues = new HashMap<>();
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        AtomicLong rowCount = new AtomicLong(1);
+        computedStatistics.forEach(statistic -> {
+            rowCount.set((long) firstNonNull(readNativeValue(BIGINT, statistic.getTableStatistics().get(ROW_COUNT), 0), 0));
+            statistic.getColumnStatistics().forEach((metadata, block) -> {
+                checkState(block.getPositionCount() == 1, "Expected a single position in aggregation result block");
+                String columnName = metadata.getColumnName();
+                Type type = types.get(columnName);
+                if (metadata.getStatisticType().equals(ColumnStatisticType.MIN_VALUE)) {
+                    // TODO overwrite, or compare?
+                    Object minimum = readNativeValue(type, block, 0);
+                    if (minimum != null) {
+                        minimums.put(columnName, minimum);
+                    }
+                }
+                else if (metadata.getStatisticType().equals(ColumnStatisticType.MAX_VALUE)) {
+                    // ignore
+                }
+                else if (metadata.getStatisticType().equals(ColumnStatisticType.NUMBER_OF_DISTINCT_VALUES)) {
+                    distinctValues.put(columnName, (long) firstNonNull(readNativeValue(BIGINT, block, 0), 0));
+                }
+                else {
+                    throw new IllegalArgumentException("Unexpected column statistic type: " + metadata.getStatisticType());
+                }
+            });
+        });
+
+        if (minimums.isEmpty()) {
+            return;
+        }
+        SchemaInfo schema = getSchema(tableName.getSchemaName());
+        double schemaMinSequenceRatio = (double) schema.properties().getOrDefault(SchemaInfo.SEQUENCE_MIN_DISTINCT_VALUES_RATIO, sequenceMinDistinctValuesRatio);
+        double tableMinSequenceRatio = (double) info.properties().getOrDefault(TableInfo.SEQUENCE_MIN_DISTINCT_VALUES_RATIO, schemaMinSequenceRatio);
+        List<String> viewColumns = columns.stream()
+                .filter(column -> !ROW_ID_COLUMN_NAME.equals(column.name()))
+                .map(column -> {
+                    String name = column.name();
+                    Type type = types.get(name);
+                    if (!type.equals(BIGINT)) {
+                        return quoted(name);
+                    }
+                    // distinct values is an approximation
+                    if ((double) distinctValues.get(name) / rowCount.get() <= tableMinSequenceRatio) {
```

Comment:
This condition might be too weak and generate false positives, I think we should also check if `max-min` is close to the number of distinct values.

Reply:
That comment didn't make a lot of sense